python/ray/serve/backend_worker.py

```diff
@@ -285,6 +296,13 @@ async def handle_request(self, request: Query) -> asyncio.Future:
 
         self.num_ongoing_requests += 1
         self.num_processing_items.set(self.num_ongoing_requests)
+
+        # Trigger a context switch so we can enqueue more requests in the
+        # meantime. Without this line and if the function is synchronous,
+        # other requests won't even get enqueued as await self.invoke_single
+        # doesn't context switch.
+        await asyncio.sleep(0)
+
         result = await self.invoke_single(request)
         self.num_ongoing_requests -= 1
         request_time_ms = (time.time() - request.tick_enter_replica) * 1000
```

Comment:
Hmm, this is kind of interesting. This implementation doesn't actually give us any guarantee on how many things get enqueued, right? What we really want here is probably an API from Ray that allows us to check the number of outstanding actor calls -- should we just add that?